docs/src/content/docs/cookbook/dexes/dedust.mdx

```diff
@@ -7,8 +7,430 @@ sidebar:
+    deadline: Int as uint32 = 0;
+    recipientAddress: Address? = null;
+    referralAddress: Address? = null;
+    fulfillPayload: Cell? = null;
+    rejectPayload: Cell? = null;
+}
+```
+
+### Swap a Jetton for another Jetton or Toncoin
+
+```tact
+/// https://docs.dedust.io/reference/tlb-schemes#message-swap-1
+message(0xe3a0d482) JettonSwapPayload {
+    // Inlined fields of SwapStep Struct
+    poolAddress: Address;
+    kind: Int as uint1 = 0;
+    limit: Int as coins = 0;
+    nextStep: SwapStep? = null;
+
+    // Set of parameters relevant for the whole swap
+    swapParams: SwapParams;
+}
+
+fun swapJetton() {
+    // NOTE: Calculate wallet address of Jetton for the target user
+    let targetJettonWalletAddress = address("");
+
+    send(SendParameters{
+        to: targetJettonWalletAddress,
+        value: ton("0.3"),
+        body: JettonTransfer{
+            // Unique identifier used to trace transactions across multiple contracts and to prevent replay attacks
+            // Defaults to 0, which means Native Vault will handle this for us
+            queryId: 0,
+            // Jetton amount for the swap
+            amount: 10, // NOTE: change to yours
+            // Address of the Jetton vault to the send message to
+            destination: address("EQDa4VOnTYlLvDJ0gZjNYm5PXfSmmtL6Vs6A_CZEtXCNICq_"),
+            // Where to return the exceeding funds
+            responseDestination: myAddress(),
+            forwardTonAmount: ton("0.25"),
+            forwardPayload: JettonSwapPayload{
+                // Address of the swap pool, which is the TON/USDT pair in this case
+                poolAddress: address("EQCxE6mUtQJKFnGfaROTKOt1lZbDiiX1kCixRv7Nw2Id_sDs"),
+                // Set of parameters relevant for the whole swap
+                swapParams: SwapParams{}, // use defaults
+            }.toCell(),
+        }.toCell(),
+    });
+}
+
+//
+// Helper Structs described earlier on this page
+//
+
+struct SwapStep {
+    poolAddress: Address;
+    kind: Int as uint1 = 0;
+    limit: Int as coins = 0;
+    nextStep: Cell?;
+}
+
+struct SwapParams {
+    deadline: Int as uint32 = 0;
+    recipientAddress: Address? = null;
+    referralAddress: Address? = null;
+    fulfillPayload: Cell? = null;
+    rejectPayload: Cell? = null;
+}
+
+//
+// Messages from the Jetton standard
+//
+
+message(0xf8a7ea5) JettonTransfer {
+    queryId: Int as uint64;
+    amount: Int as coins;
+    destination: Address;
+    responseDestination: Address;
+    customPayload: Cell? = null;
+    forwardTonAmount: Int as coins;
+    forwardPayload: Cell?; // slightly adjusted
+}
+```
+
+## Liquidity Provisioning
+
+To provide liquidity to a particular DeDust pool, you must provide both assets. The pool will then issue special _LP tokens_ to the depositor's address.
+
+Due to the limitations of TON, it is not possible to perform two simultaneous transfers. Therefore, the balances of user deposits must be held until LP tokens are distributed.
```

Comment:
I'd rewrite or remove this statement.

both `not possible to perform two simultaneous transfers` and `the balances of user deposits must be held until LP tokens are distributed` are misleading